pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -636,48 +628,10 @@ func (a *Actuator) ensureProviderID(ctx context.Context, machine *machinev1beta1
 	return nil
 }
 
-// handleNodeFinalizer adds finalizer to Node if not already exists
-// it also store the node annotations and labels on Machine annotations
-// upon node deletion
-func (a *Actuator) handleNodeFinalizer(ctx context.Context, machine *machinev1beta1.Machine) error {
-	node, err := a.getNodeByMachine(ctx, machine)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
-
-		log.Printf("Failed to find node associated with Machine %s, error: %s", machine.Name, err.Error())
-		return err
-	}
-
-	if node.DeletionTimestamp.IsZero() {
-		if !utils.StringInList(node.Finalizers, nodeFinalizer) {
-			//add finalizer
-			node.Finalizers = append(node.Finalizers, nodeFinalizer)
-			if err := a.client.Update(ctx, node); err != nil {
-				log.Printf("Failed to add node finalizer to node %s, error: %s", node.Name, err.Error())
-				return err
-			}
-		}
-	} else {
-		//node is in deletion process
-		if utils.StringInList(node.Finalizers, nodeFinalizer) {
-			if err := a.storeAnnotationsAndLabels(ctx, node, machine); err != nil {
-				return err
-			}
-
-			//remove finalizer
-			node.Finalizers = utils.FilterStringFromList(node.Finalizers, nodeFinalizer)
-			if err := a.client.Update(ctx, node); err != nil {
-				log.Printf("Failed to remove node finalizer from %s, error: %s", node.Name, err.Error())
-				return err
-			}
-		}
-	}
-	return nil
-}
-
 // removeNodeFinalizer removes the finalizer from the Node
+// We don't add a finalizer any more unless we are about to delete the Node
+// during remediation, but a previous version of the actuator may have left one
```

Comment:
If we don't remove existing finalizers, we're still stuck with the problem of immortable node, until they get remediation.
So if a machine enters a `Failed` state, the finalizer won't be removed